script.js

```diff
@@ -1,46 +1,221 @@
+    ru.classList.add("ru");
+
+    const ruKeyCaseUp = document.createElement("div");
+
+    ruKeyCaseUp.classList.add("caseUp");
+    ruKeyCaseUp.innerText = ruCaseUp[n][i];
+    ru.append(ruKeyCaseUp);
+
+    const ruKeyCaseDown = document.createElement("div");
+    ruKeyCaseDown.classList.add("caseDown");
+    ruKeyCaseDown.innerText = ruCaseDown[n][i];
+    ru.append(ruKeyCaseDown);
+
+
+    const en = document.createElement("div");
+    en.classList.add("en");
+
+    const enKeyCaseUp = document.createElement("div");
+    enKeyCaseUp.classList.add("caseUp");
+    enKeyCaseUp.innerText = enCaseUp[n][i];
+    en.append(enKeyCaseUp);
+
+    const enKeyCaseDown = document.createElement("div");
+    enKeyCaseDown.classList.add("caseDown");
+    enKeyCaseDown.innerText = enCaseDown[n][i];
+    en.append(enKeyCaseDown);
+
+    key.append(ru);
+    key.append(en);
+    row.append(key);
+  }
+}
+// add rows to keyboard
+/* eslint no-plusplus: ["error", { "allowForLoopAfterthoughts": true }] */
+for (let i = 0; i < 5; i += 1) {
+  const row = document.createElement("div");
+  row.className = "row";
+  addKeys(row, i);
+  keyboard.append(row);
+}
+// UpperCase and Language
+const enKeys = document.querySelectorAll(".en");
+const ruKeys = document.querySelectorAll(".ru");
+const upKeys = document.querySelectorAll(".caseUp");
+const downKeys = document.querySelectorAll(".caseDown");
+
+upKeys.forEach((el) => el.classList.toggle("hidden"));
+
+function changeLang() {
+  let lang = localStorage.getItem("lang");
+  if (!lang) lang = "ru";
+  if (lang === "ru") {
+    enKeys.forEach((el) => el.classList.remove("hidden"));
+    ruKeys.forEach((el) => el.classList.add("hidden"));
+    localStorage.setItem("lang", "en");
+  } else {
+    enKeys.forEach((el) => el.classList.add("hidden"));
+    ruKeys.forEach((el) => el.classList.remove("hidden"));
+    localStorage.setItem("lang", "ru");
+  }
+}
+function hideLangKeys() {
+  let lang = localStorage.getItem("lang");
+  if (!lang) lang = "ru";
+  if (lang === "ru") {
+    enKeys.forEach((el) => el.classList.add("hidden"));
+  } else {
+    ruKeys.forEach((el) => el.classList.add("hidden"));
+  }
+}
+hideLangKeys();
+function caps() {
+  upKeys.forEach((el) => el.classList.toggle("hidden"));
+  downKeys.forEach((el) => el.classList.toggle("hidden"));
+}
+// event
+const keyControls = ["Backspace", "Tab", "CapsLock", "ShiftLeft", "ControlLeft", "MetaLeft",
+  "AltLeft", "Space", "AltRight", "ControlRight", "ShiftRight", "Enter", "Delete", "Backspace"];
+
+function printKey(elem) {
+  function findKey(el) {
+    for (let i = 0; i < el.children.length; i += 1) {
+      const str = el.children[i].classList.value;
+      if (str.indexOf("hidden") === -1) {
+        findKey(el.children[i]);
+      }
+    }
+    return elem.innerText;
+  }
+  let returnPointer;
+  if (keyControls.indexOf(elem.classList[1]) === -1) {
+    const char = findKey(elem);
+    screen.value += char;
+  } else {
+    switch (elem.classList[1]) {
```

Comment:
it's not clear here why index 1 is used. Classes order is not really reliable. I would suggest to add custom attribute to element additionally to class.

Reply:
because the first index stores a classname with the same name as keycode, I should have created a variable for clarity((

Reply:
My main idea is to not rely on classes at all when you need to save data. 
In this particular case this app is pretty small and you work on it alone. In real apps lot of people work on the same codebase and order of classes will be changed for sure.

There're `setAttribute`/`getAttribute` functions on element which you can use to set your custom attribute, e.g. `keycode`. And use it wherever you want.